scripts/nlp_language_modeling/merge_lora_weights/merge.py

```diff
@@ -0,0 +1,223 @@
+            trainer=trainer,
+            return_config=True,
+            save_restore_connector=save_restore_connector,
+        )
+        OmegaConf.set_struct(pretrained_cfg, True)
+        with open_dict(pretrained_cfg):
+            pretrained_cfg.sequence_parallel = False
+            pretrained_cfg.activations_checkpoint_granularity = None
+            pretrained_cfg.activations_checkpoint_method = None
+            pretrained_cfg.precision = trainer.precision
+            if trainer.precision == "16":
+                pretrained_cfg.megatron_amp_O2 = False
+        model = MegatronGPTModel.restore_from(
+            restore_path=cfg.gpt_model_file,
+            trainer=trainer,
+            override_config_path=pretrained_cfg,
+            map_location=torch.device("cpu"),
+            save_restore_connector=save_restore_connector,
+        )
+    elif cfg.checkpoint_dir:
+        app_state = AppState()
+        if cfg.tensor_model_parallel_size > 1 or cfg.pipeline_model_parallel_size > 1:
+            app_state.model_parallel_size = cfg.tensor_model_parallel_size * cfg.pipeline_model_parallel_size
+            app_state.tensor_model_parallel_size = cfg.tensor_model_parallel_size
+            app_state.pipeline_model_parallel_size = cfg.pipeline_model_parallel_size
+            (
+                app_state.tensor_model_parallel_rank,
+                app_state.pipeline_model_parallel_rank,
+                app_state.model_parallel_size,
+                app_state.data_parallel_size,
+                app_state.pipeline_model_parallel_split_rank,
+                app_state.virtual_pipeline_model_parallel_rank,
+            ) = fake_initialize_model_parallel(
+                world_size=app_state.model_parallel_size,
+                rank=trainer.global_rank,
+                tensor_model_parallel_size_=cfg.tensor_model_parallel_size,
+                pipeline_model_parallel_size_=cfg.pipeline_model_parallel_size,
+                pipeline_model_parallel_split_rank_=cfg.pipeline_model_parallel_split_rank,
+            )
+        checkpoint_path = inject_model_parallel_rank(os.path.join(cfg.checkpoint_dir, cfg.checkpoint_name))
+        model = MegatronGPTModel.load_from_checkpoint(checkpoint_path, hparams_file=cfg.hparams_file, trainer=trainer)
+    else:
+        raise ValueError("need at least a nemo file or checkpoint dir")
+
+    lora_model_cfg = MegatronGPTLoRAModel.restore_from(
+        restore_path=cfg.lora_model_path, trainer=trainer, return_config=True,
+    )
+
+    # load the lora weights on cpu for all ranks of the lora model
+    lora_weights = load_lora(cfg.lora_model_path, model.cfg.tensor_model_parallel_size)
+
+    # merge the lora weights with the base model, for this current rank.
+    merged_weights = merge(
+        model.state_dict(),
+        lora_weights,
+        tp=model.cfg.tensor_model_parallel_size,
+        num_layers=model.cfg.num_layers,
+        curr_rank=model.global_rank,
+    )
+
```

Comment:
wouldn't we need to remove `self.adapter_layer`? otherwise both updated qkv and adapter will be used?

Reply:
okay, nvm, the base should be fine